sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/session.py

```diff
@@ -181,7 +182,8 @@ def _incoming_attach(self, frame):
             self._output_handles[outgoing_handle] = new_link
             self._input_handles[frame[1]] = new_link
         except ValueError:
-            pass  # TODO: Reject link
+            # Reject Link
+            new_link.detach()
```

Comment:
it looks like this new link is being defined in the except KeyError block, but being detached here where it wouldn't exist unless I'm missing something. Do we want to indent this and do another try/except in the KeyError block? Or, more likely, this will be a ValueError from the try on line 172, so should this rejection be handled some other way?

Reply:
good catch !! now its rejecting the right link :)